packages/discord.js/src/structures/SoundboardSound.js

```diff
@@ -0,0 +1,198 @@
+    super(client);
+
+    /**
+     * The id of the soundboard sound
+     * @type {Snowflake|string}
+     */
+    this.soundId = data.sound_id;
+
+    this._patch(data);
+  }
+
+  _patch(data) {
+    if ('available' in data) {
+      /**
+       * Whether this soundboard sound is available
+       * @type {?boolean}
+       */
+      this.available = data.available;
+    } else {
+      this.available ??= null;
+    }
+
+    if ('name' in data) {
+      /**
+       * The name of the soundboard sound
+       * @type {?string}
+       */
+      this.name = data.name;
+    } else {
+      this.name ??= null;
+    }
+
+    if ('volume' in data) {
+      /**
+       * The volume of the soundboard sound, from 0 to 1
+       * @type {?number}
+       */
+      this.volume = data.volume;
+    } else {
+      this.volume ??= null;
+    }
+
+    if ('emoji_id' in data) {
+      /**
+       * The emoji id of the soundboard sound
+       * @type {?Snowflake}
+       */
+      this.emojiId = data.emoji_id;
+    } else {
+      this.emojiId ??= null;
+    }
+
+    if ('emoji_name' in data) {
+      /**
+       * The emoji name of the soundboard sound
+       * @type {?string}
+       */
+      this.emojiName = data.emoji_name;
+    } else {
+      this.emojiName ??= null;
+    }
+
+    if ('guild_id' in data) {
+      /**
+       * The guild id of the soundboard sound
+       * @type {?Snowflake}
+       */
+      this.guildId = data.guild_id;
+    } else {
+      this.guildId ??= null;
+    }
+
+    if ('user' in data) {
+      /**
+       * The user who created this soundboard sound
+       * @type {?User}
+       */
+      this.user = this.client.users._add(data.user);
+    } else {
+      this.user ??= null;
+    }
+  }
+
+  /**
+   * The guild this soundboard sound is part of
+   * @type {?Guild}
+   * @readonly
+   */
+  get guild() {
+    return this.client.guilds.resolve(this.guildId);
+  }
+
+  /**
+   * The timestamp the soundboard sound was created at
+   * @type {number}
+   * @readonly
+   */
+  get createdTimestamp() {
+    return DiscordSnowflake.timestampFrom(this.soundId);
+  }
+
+  /**
+   * The time the sticker was created at
+   * @type {Date}
+   * @readonly
+   */
+  get createdAt() {
+    return new Date(this.createdTimestamp);
+  }
+
+  /**
+   * A link to the soundboard sound
+   * @type {string}
+   * @readonly
+   */
+  get url() {
+    return this.client.rest.cdn.soundboardSound(this.soundId);
+  }
+
+  /**
+   * Edits the soundboard sound.
+   * @param {GuildSoundboardSoundEditOptions} options The options to provide
+   * @returns {Promise<SoundboardSound>}
+   * @example
+   * // Update the name of a soundboard sound
+   * soundboardSound.edit({ name: 'new name' })
+   *   .then(sound => console.log(`Updated the name of the soundboard sound to ${sound.name}`))
+   *   .catch(console.error);
+   */
+  edit(options) {
+    if (!this.guild) throw new DiscordjsError(ErrorCodes.NotGuildSoundboardSound, 'edited');
```

Comment:
This will be a sync error, either mark the function as async or wrap in Promise.reject